### ISSUE TITLE
i3xrocks icons not displayed properly

### ISSUE THREAD

**Issue by naib864** [gardotd426/regolith-de#6]
Just started testing this on my t61 running manjaro and I noticed that the material design icons aren't displayed properly, even though I installed the font manually.

I think in the font version I'm using the characters are at another position in the character table than on the font that regolith on uses on ubuntu.
Is there a font on arch that will make the icons work?

**Comment by hughesjs** [comment]
I am getting similarly bizarre icons as @naib864.

**Comment by gardotd426** [comment]
Some of those icons are clearly correct, but some of them seem like they're surely not. But without having an actual regolith install in front of me, it's hard to tell which is which. The thing is, it looks like the only fonts Regolith includes are a couple regular fonts. I'm going to boot into a regolith ISO and see what I can figure out. 

@hughesjs do me a favor and try removing ttf-font-awesome and install a nerd font instead, and reboot/log out and back in

**Comment by hughesjs** [comment]
Had to do the uninstall with `-Rdd` and installed the nerd font.

Everything still looks weird.

**Comment by gardotd426** [comment]
Okay, I'll see what I can figure out.

**Comment by gardotd426** [comment]
@naib864 @hughesjs I figured it out (effectively). 

So, the deal is, regolith keeps all of it's configs in a special ~/.Xresources file, called ~/.Xresources-regolith. All it does, is it points to whichever style the user has chosen. It comes with like 7 or something. Well, the default one is lascaille. I find the colorscheme gorgeous, and it's the default on Regolith itself, so I kept it. Well, the i3xrocks font config file for lascaille has all the goofed up icons. I don't know why, because I checked the font-awesome version that Regolith uses, and the character maps match up, meaning the icon in the config file, for example the Windows logo, that's U+F17A, which is also the icon at U+F17A in the Font Awesome version Regolith itself uses. So it honestly seems that even on Regolith, the default font should look like that. Something else is causing it to fix itself. 

But none of that matters to us. All the other looks use the exact same font config for the actual i3xrocks bar. So all I have to do is use one of those typeface config files instead, and we're good to go. I'm working on it now, I'll comment here when it's patched so you guys can pull and reinstall.

**Comment by gardotd426** [comment]
@naib864 @hughesjs It's fixed, can you clone a fresh repo, reinstall, and log out and back in and let me know what's up?

**Comment by hughesjs** [comment]
@gardotd426  good work dude! Yeah, that's looking a hell of a lot more sane now...

I'll have to do some digging wrt i3xrocks to find out why I've got @a(ss) [] popping up on the reg though...

The update also overwrote my config file, I'd suggest we allow all config files to be overridden from ~/.config/regolith-de/{i3, i3xrocks...} to stop this.

Then again, that seems like an upstream thing to me.

**Comment by gardotd426** [comment]
>I'll have to do some digging wrt i3xrocks to find out why I've got @A(ss) [] popping up on the reg though...

What do you mean? The icons still aren't showing? Can I see screenshots? They all show up here. 

>The update also overwrote my config file, I'd suggest we allow all config files to be overridden from ~/.config/regolith-de/{i3, i3xrocks...} to stop this.

Can't. Packages are never allowed to touch user home directory files, so no package can ever write to ~/.config/anything. 

However, Regolith DE **does** read from ~/.config/regolith/whatever. For example, if ~/.config/regolith/i3/config exists, it will read that ***INSTEAD*** of /etc/regolith/i3/config. That's the exact same way regular i3 works. i3, i3-gaps, etc. all provide /etc/i3/config as the default, and the user is supposed to copy that to ~/.config/i3/config and make changes to THAT file. /etc/i3/config gets overwritten on every update just like /etc/regolith/i3/config will.

Just imagine it like this: everything in /etc/regolith can also be placed in ~/.config/regolith, and Regolith will read from ~/.config/regolith first.

**Comment by gardotd426** [comment]
@naib864, if you could please pull the changes and update, so I can confirm the issue is fixed and close this.

**Comment by hughesjs** [comment]
@gardotd426 that config file behaviour is exactly what I meant in my comment. Didn't realise it still did that so that's good news.

As for the weird text on the bar... See the @a(ss) on the left.
![image](https://user-images.githubusercontent.com/4203766/91399442-e6202b00-e835-11ea-896a-ec29895b4dac.png)

**Comment by gardotd426** [comment]
Okay, I'll look into it tomorrow (it's really late on the east coast), I'll post whatever I find or don't find.

**Comment by hughesjs** [comment]
Thanks mate, currently working offshore so no time to dig myself.

Sent from ProtonMail mobile

-------- Original Message --------

**Comment by naib864** [comment]
Sorry, somehow didn't see the notifications for your comments. Will check if the patch works this evening.

**Comment by naib864** [comment]
It looks like this now. Symbols are there but the @a(ss) is weird and the spacing is weird as well.
![Bildschirmfoto von 2020-08-29 16-44-40](https://user-images.githubusercontent.com/66312463/91636504-75c60500-ea01-11ea-976d-cc2a83afa178.png)

**Comment by naib864** [comment]
Dunno what changed, but now i3xrocks is completely gone. Did a fresh install of manjaro-i3 and the first thing I did was installing regolith. Theme isn't lascaille, which is weird too. Different conf doesn't change that, the bar is empty.
![Screenshot from 2020-09-01 01-22-08](https://user-images.githubusercontent.com/66312463/91770858-44585f80-ebe2-11ea-9c13-49633e920e29.png)

**Comment by gardotd426** [comment]
What do you mean you tried it with a live USB? You booted up a live USB of Manjaro and installed regolith on it, and logged out and back in? 

Did you log in to "Regolith" or "i3" when you chose your desktop environment?

**Comment by gardotd426** [comment]
>Different conf doesn't change that, the bar is empty.

user config files go in ~/.config/regolith/ just like the setup of /etc/regolith. Basically, you'll want to run `cp -r /etc/regolith/* ~/.config/regolith/` and edit the `.config/regolith/whatever/config` files.

**Comment by naib864** [comment]
I log into Regolith and I used both the default config and my own custom config that used to work in `~/.config/regolith/i3`.

I installed regolith in a live session of manjaro i3, logged out and relogged into the session selecting regolith at the login screen.

**Comment by gardotd426** [comment]
What do you mean when you say the default theme isn't lascaille? Is that just based on look or are you actually seeing the theme and it says something other than lascaille? 

Nothing has changed about the PKGBUILD so I'm having trouble wondering what could have broken

**Comment by naib864** [comment]
Well, the theme composition doesn't match the one I've seen before. Gnome theme is set different and the color scheme is different as well. Setting the theme to lascaille by using `regolith-look set lascaille` and then `regolith-look refresh` doesn't change it either.

**Comment by gardotd426** [comment]
@naib864 I can't reproduce on Manjaro KDE. 

I initially tried to use Manjaro i3's iso, but it crashed consistently at the same spot during installation in a VM, which isn't surprising considering it's a community edition and not an official edition. 

So I downloaded Manjaro KDE, installed it in a VM, booted into it, installed regolith (and nothing else, aside from updating the system first), and logged out and in to Regolith. Everything works exactly as it should....

![image](https://user-images.githubusercontent.com/54234607/91920378-d91c9500-ec96-11ea-88ca-347a7f3f6f59.png)

(note the A(ss) is also gone, I haven't changed the PKGBUILD yet to fix it, I'll do that in a bit, but yeah I know how to fix it)

Manjaro i3 (and KDE edition) comes with like 570-something needed updates, did you update after installation? I suggest doing that. 

Beyond that, I'm not sure how to help you figure out your issue without some help on your end, I've never seen this in the 7 or 8 fresh installations I've made.
